service/src/main/java/com/basis/grupoum/sgt/service/servico/OfertaServico.java

```diff
@@ -0,0 +1,148 @@
+        Situacao situacao = new Situacao();
+        situacao.setId(idSituacao);
+        List<Oferta> ofertasPorSitucao = ofertaRepositorio.findAllBySituacao(situacao);
+        return ofertaListagemMapper.toDto(ofertasPorSitucao);
+    }
+
+    public OfertaDTO obterPorId(Long id){
+        Oferta oferta = ofertaRepositorio
+                .findById(id).orElseThrow(() -> new RegraNegocioException("Oferta não encontrada"));
+        return ofertaMapper.toDto(oferta);
+    }
+
+    public OfertaDTO salvar(OfertaDTO ofertaDTO){
+        ofertaDTO = alteraDisponibilidadeItensOfertados(ofertaDTO, false);
+        Oferta oferta = ofertaMapper.toEntity(ofertaDTO);
+        ofertaRepositorio.save(oferta);
+
+        emailServico.sendEmail(criarEmailOferta(oferta));
+        return ofertaMapper.toDto(oferta);
+    }
+
+    public OfertaDTO atualizar(OfertaDTO ofertaDTO){
+        Oferta oferta = ofertaMapper.toEntity(ofertaDTO);
+        ofertaRepositorio.save(oferta);
+        return ofertaMapper.toDto(oferta);
+    }
+
+    public void deletar(Long idOferta){
+        ofertaRepositorio.deleteById(idOferta);
+    }
+
+    public void aceitar(Long idOferta){
+        OfertaDTO ofertaDTO = obterPorId(idOferta);
+        ItemDTO itemAuxDTO = itemServico.obterPorId(ofertaDTO.getItemDtoId());
+
+        Long idUsuarioOfertado = itemAuxDTO.getUsuarioDtoId();
+
+        itemAuxDTO.setUsuarioDtoId(ofertaDTO.getUsuarioDtoId());
+        itemServico.atualizar(itemAuxDTO);
+
+        ofertaDTO.getItensOfertados().forEach(itemDTO -> {
+            itemDTO.setUsuarioDtoId(idUsuarioOfertado);
+            itemDTO.setDisponibilidade(true);
+            itemServico.atualizar(itemDTO);
+        });
+
+        Item itemCancelado = itemMapper.toEntity(itemServico.obterPorId(ofertaDTO.getItemDtoId()));
+        List<OfertaDTO> ofertasCanceladas = ofertaMapper.toDto(ofertaRepositorio.findAllByItem(itemCancelado));
+        cancelaDemaisOfertas(ofertasCanceladas,ofertaDTO.getItemDtoId());
+
+        ofertaDTO.setSituacaoDtoId(2L);
+        atualizar(ofertaDTO);
+    }
+
+    public void recusar(Long idOferta){
+        OfertaDTO ofertaDTO = obterPorId(idOferta);
+        ofertaDTO.setSituacaoDtoId(3L);
+        ofertaDTO = alteraDisponibilidadeItensOfertados(ofertaDTO, true);
+        atualizar(ofertaDTO);
+    }
+
+    public void cancelar(Long idItem){
+        Item itemCancelado = itemMapper.toEntity(itemServico.obterPorId(idItem));
+        List<OfertaDTO> ofertasCanceladas = ofertaMapper.toDto(ofertaRepositorio.findAllByItem(itemCancelado));
+        cancelaDemaisOfertas(ofertasCanceladas,idItem);
+    }
+
+    public OfertaDTO alteraDisponibilidadeItensOfertados(OfertaDTO ofertaDTO, boolean disponibilidade){
+        ofertaDTO.getItensOfertados().forEach(itemDTO -> {
+            itemDTO = itemServico.obterPorId(itemDTO.getId());
+            itemDTO.setDisponibilidade(disponibilidade);
+            itemServico.atualizar(itemDTO);
```

Comment:
seu método atualizar chama um repositorio, nunca podemos chamar um repositorio dentro de um for/foreach, criar um método atualizarTodas, que vai ter um saveAll(lista);
e chamar o saveAll() fora do foreach, ou seja, armazenar essa lista de ofertas em uma variavel e chamar ela no saveAll()